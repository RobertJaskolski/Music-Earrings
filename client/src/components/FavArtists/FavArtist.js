import React from "react";
import { Div, Span, P } from "./style/style";
import QueueIcon from "@material-ui/icons/Queue";

function FavArtist(props) {
  const { artist, changeFav, addToFilters } = props;
  const { name, images } = artist;
  return (
    <Div desktop={changeFav}>
      <Span>
        <img
          alt={name}
          src={images[2]?.url || "/images/wrapper.jpg"}
          width={50}
          height={50}
        />
      </Span>
      <Span>
        <P>{name}</P>
      </Span>
      <Span>
        <QueueIcon onClick={() => addToFilters(artist)} />
      </Span>
    </Div>
  );
}

export default FavArtist;
