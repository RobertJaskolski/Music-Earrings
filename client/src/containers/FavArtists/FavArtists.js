// Import outside
import React from 'react';
import { Grid } from '@material-ui/core';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import { useTranslation } from 'react-i18next';
// Import Components
import FavArtist from '../../components/FavArtists/FavArtist';
// Import Styles
import { Div, H2, GridContainer } from './style/style';

function FavArtists({
  auth,
  userFavoriteArtists,
  artistSeeds,
  trackSeeds,
  addArtistToSeeds,
  handleOpenFail,
  handleOpenSuccessArtist,
}) {
  const { t } = useTranslation();
  const changeFav = useMediaQuery('(min-width:500px)');
  const addToFilters = (item) => {
    if (trackSeeds.length + artistSeeds.length < 5) {
      addArtistToSeeds(item);
      handleOpenSuccessArtist();
    } else {
      handleOpenFail();
    }
  };
  if (!auth) {
    return null;
  }
  return (
    <GridContainer container>
      <Grid item xs={12}>
        <H2>{t('containers.FavArtists.yoursfavs')}</H2>
        <Div>
          {userFavoriteArtists?.map((item) => {
            return (
              <FavArtist
                addToFilters={addToFilters}
                changeFav={changeFav}
                key={item.id}
                artist={item}
              />
            );
          })}
        </Div>
      </Grid>
    </GridContainer>
  );
}

export default FavArtists;
