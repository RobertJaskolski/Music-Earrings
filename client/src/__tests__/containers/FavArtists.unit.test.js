import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FavArtists from '../../containers/FavArtists/FavArtists';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

describe('Fav artists container', () => {
  test('should not render if user logout', () => {
    const wrapper = shallow(<FavArtists auth={false} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test('should render correctly when user login', () => {
    const wrapper = shallow(<FavArtists auth={true} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
