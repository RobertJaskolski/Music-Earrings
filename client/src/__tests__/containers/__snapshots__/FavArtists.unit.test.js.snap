// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fav artists container should not render if user logout 1`] = `""`;

exports[`Fav artists container should render correctly when user login 1`] = `
<Styled(WithStyles(ForwardRef(Grid)))
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={12}
  >
    <styled.h2>
      containers.FavArtists.yoursfavs
    </styled.h2>
    <styled.div />
  </WithStyles(ForwardRef(Grid))>
</Styled(WithStyles(ForwardRef(Grid)))>
`;
